daemon/cmd/daemon.go

```diff
@@ -1148,6 +1168,43 @@ func (d *Daemon) bootstrapClusterMesh(nodeMngr *nodemanager.Manager) {
 	bootstrapStats.clusterMeshInit.End(true)
 }
 
+// ReloadOnDeviceChange regenerates device related information and reloads the datapath.
+func (d *Daemon) ReloadOnDeviceChange(devices []string) {
+	log.WithField(logfields.Devices, devices).Infof("Datapath reload initiated due to changed devices.")
+
+	option.Config.Devices = devices
+
+	if option.Config.EnableIPv4Masquerade && option.Config.EnableBPFMasquerade {
+		if err := node.InitBPFMasqueradeAddrs(devices); err != nil {
+			log.Warnf("InitBPFMasqueradeAddrs failed: %s", err)
+		}
+	}
+
+	if option.Config.EnableNodePort {
+		if err := node.InitNodePortAddrs(devices, option.Config.LBDevInheritIPAddr); err != nil {
+			log.WithError(err).Warn("Failed to initialize NodePort addresses")
+		} else {
+			// Synchronize services and endpoints to reflect new addresses onto lbmap.
+			d.svc.SyncServicesOnDeviceChange(d.Datapath().LocalNodeAddressing())
+			d.syncEndpointsAndHostIPs()
+		}
+	}
+
+	// Recreate node_config.h to reflect the mac addresses of the new devices.
+	if err := d.createNodeConfigHeaderfile(); err != nil {
+		log.WithError(err).Warn("Failed to re-create node config header")
+		return
+	}
+
+	// Reload the datapath.
+	wg, err := d.TriggerReloadWithoutCompile("devices changed")
```

Comment:
One more: Could this cause two goroutines to run the `Reinitialize()` of the datapath concurrently? Is that a problem?

Reply:
OK, quick answer: No, it has a lock. But you'd do two back-to-back reinitalizations even if the first one successfully handles all the new devices. :shrug: I guess that's fine.

Feels a bit like we need an eventqueue here too, but I think that's more of an optimization rather than a necessary step to support this new functionality. I think we can get away without any further changes to this specific part for now.